app/lib/tweet_transformer.rb

```diff
@@ -19,6 +19,13 @@ def self.replace_links(text, urls)
 
   def self.replace_mentions(text)
     twitter_mention_regex = /(\s|^.?|[^A-Za-z0-9_!#\$%&*@＠\/])([@＠][A-Za-z0-9_]+)(?=[^A-Za-z0-9_@＠]|$)/
+
+    if Rails.configuration.x.translate_twitter_accounts
+      mapping = User.twitter_mastodon_mapping
+      mapping.default_proc = proc {|hash, key| key }
+      text.gsub!(twitter_mention_regex, mapping)
```

Comment:
This does not work currently. `twitter_mention_regex` also matches leading spaces. However, the mapping hash does not have this leading space and thus does not carry out a substitution.

@renatolond What is the best approach here? Add a different twitter regexp?

Reply:
I don't remember the mention results exactly, but I think you have the leading spaces in `\1`, so you could something like:
```
gsub!(..., "\1#{mapping}")
```
since you're using a proc there, I'm not too sure on how you could do that, though :thinking:

Reply:
Hi! `mapping` is a hash of the form `{"@twitter_handle" => "@mastodon_handle@server.com", ...}`. gsub can pick here the replacement text if the capture is found amongst the hash keys. The default_proc ensures that twitter mentions not found in the mapping are just replaced by themselves, i.e. not modified.

The expression `"\1#{mapping}"` would be converted into a string and gsub would then replace all captures by the same text.

If we can rewrite the `twitter_mention_regex` to not match a leading space/char, then we could use the regexp. I assume though that there was a specific reason to match a leading space…

Reply:
Hmm, and wouldn't it work if you do a transform_keys in your `mapping` hash to have `\1@twitter_handle` keys instead?